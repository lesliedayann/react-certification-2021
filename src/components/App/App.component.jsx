import React, { useLayoutEffect } from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';

import AuthProvider from '../../providers/Auth';
import HomePage from '../../pages/Home';
import LoginPage from '../../pages/Login';
import NotFound from '../../pages/NotFound';
import Video from '../../pages/VideoTemplate/Video.page';
import Navbar from '../Navabar';
import Layout from '../Layout';
import { random } from '../../utils/fns';
import { AppContext, useAppContextValue } from '../../utils/AppContext';

function App() {
  const appContextValue = useAppContextValue();

  useLayoutEffect(() => {
    const { body } = document;

    function rotateBackground() {
      const xPercent = random(100);
      const yPercent = random(100);
      body.style.setProperty('--bg-position', `${xPercent}% ${yPercent}%`);
    }

    const intervalId = setInterval(rotateBackground, 3000);
    body.addEventListener('click', rotateBackground);

    return () => {
      clearInterval(intervalId);
      body.removeEventListener('click', rotateBackground);
    };
  }, []);

  return (
    <AppContext.Provider value={appContextValue}>
      <BrowserRouter>
        <AuthProvider>
          <Layout>
            <div style={{ height: '100%', width: '100%' }}>
              <Navbar />
              <Switch>
                <Route exact path="/">
                  <HomePage />
                </Route>
                <Route exact path="/login">
                  <LoginPage />
                </Route>
                <Route path="/:id" component={Video} />
                <Route path="*">
                  <NotFound />
                </Route>
              </Switch>
            </div>
          </Layout>
        </AuthProvider>
      </BrowserRouter>
    </AppContext.Provider>
  );
}

export default App;
