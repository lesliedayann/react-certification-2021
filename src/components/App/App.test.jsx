import React from 'react';
import { render } from '@testing-library/react';
import App from './App.component';

describe('Testing App component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = render(<App />);
  });

  it('should to take snapshop', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
