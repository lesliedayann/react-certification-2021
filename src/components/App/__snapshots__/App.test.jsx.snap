// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App component should to take snapshop 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main>
        <div
          style="height: 100%; width: 100%;"
        >
          <div
            class="makeStyles-grow-1"
          >
            <header
              class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-colorPrimary-3 MuiPaper-elevation4"
            >
              <div
                class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
              >
                <button
                  aria-label="open drawer"
                  class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-4 MuiIconButton-colorInherit MuiIconButton-edgeStart"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <div
                  class="makeStyles-search-6"
                >
                  <div
                    class="makeStyles-searchIcon-7"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiInputBase-root makeStyles-inputRoot-8"
                  >
                    <input
                      aria-label="search"
                      class="MuiInputBase-input makeStyles-inputInput-9"
                      placeholder="Search…"
                      type="text"
                      value="wizeline"
                    />
                  </div>
                </div>
                <div
                  class="makeStyles-space-2"
                />
                <div
                  class="makeStyles-sectionDesktop-10"
                >
                  <div
                    class="MuiFormGroup-root"
                  >
                    <label
                      class="MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiSwitch-root"
                      >
                        <span
                          aria-disabled="false"
                          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-15 MuiSwitch-switchBase WithStyles(ForwardRef(Switch))-switchBase-12 MuiSwitch-colorSecondary"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <input
                              class="PrivateSwitchBase-input-18 MuiSwitch-input"
                              id="customSwitch"
                              name="customSwitch"
                              type="checkbox"
                              value=""
                            />
                            <span
                              class="MuiSwitch-thumb"
                            />
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </span>
                        <span
                          class="MuiSwitch-track WithStyles(ForwardRef(Switch))-track-14"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiFormControlLabel-label makeStyles-label-11 MuiTypography-body1"
                      >
                        Dark Mode
                      </span>
                    </label>
                  </div>
                  <button
                    aria-controls="primary-search-account-menu"
                    aria-haspopup="true"
                    aria-label="account of current user"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeEnd"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </header>
          </div>
          <section
            class="homepage"
          >
            <h1>
              Welcome to the Challenge!
            </h1>
            <div
              class="sc-bdnxRM bSfclB"
            />
          </section>
        </div>
      </main>
    </div>
    <div
      aria-hidden="true"
      class="MuiPopover-root"
      id="primary-search-account-menu"
      role="presentation"
      style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px; visibility: hidden;"
    >
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiPaper-root MuiMenu-paper MuiPopover-paper MuiPaper-elevation8 MuiPaper-rounded"
        style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
        tabindex="-1"
      >
        <ul
          class="MuiList-root MuiMenu-list MuiList-padding"
          role="menu"
          tabindex="-1"
        >
          <li
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
            role="menuitem"
            tabindex="0"
          >
            Iniciar sesion
            <span
              class="MuiTouchRipple-root"
            />
          </li>
        </ul>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div>
    <main>
      <div
        style="height: 100%; width: 100%;"
      >
        <div
          class="makeStyles-grow-1"
        >
          <header
            class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-colorPrimary-3 MuiPaper-elevation4"
          >
            <div
              class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
            >
              <button
                aria-label="open drawer"
                class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-4 MuiIconButton-colorInherit MuiIconButton-edgeStart"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <div
                class="makeStyles-search-6"
              >
                <div
                  class="makeStyles-searchIcon-7"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiInputBase-root makeStyles-inputRoot-8"
                >
                  <input
                    aria-label="search"
                    class="MuiInputBase-input makeStyles-inputInput-9"
                    placeholder="Search…"
                    type="text"
                    value="wizeline"
                  />
                </div>
              </div>
              <div
                class="makeStyles-space-2"
              />
              <div
                class="makeStyles-sectionDesktop-10"
              >
                <div
                  class="MuiFormGroup-root"
                >
                  <label
                    class="MuiFormControlLabel-root"
                  >
                    <span
                      class="MuiSwitch-root"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-15 MuiSwitch-switchBase WithStyles(ForwardRef(Switch))-switchBase-12 MuiSwitch-colorSecondary"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <input
                            class="PrivateSwitchBase-input-18 MuiSwitch-input"
                            id="customSwitch"
                            name="customSwitch"
                            type="checkbox"
                            value=""
                          />
                          <span
                            class="MuiSwitch-thumb"
                          />
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiSwitch-track WithStyles(ForwardRef(Switch))-track-14"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiFormControlLabel-label makeStyles-label-11 MuiTypography-body1"
                    >
                      Dark Mode
                    </span>
                  </label>
                </div>
                <button
                  aria-controls="primary-search-account-menu"
                  aria-haspopup="true"
                  aria-label="account of current user"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeEnd"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </header>
        </div>
        <section
          class="homepage"
        >
          <h1>
            Welcome to the Challenge!
          </h1>
          <div
            class="sc-bdnxRM bSfclB"
          />
        </section>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
